packages/firestore/src/core/query.ts

```diff
@@ -349,14 +349,16 @@ export class Query {
     return null;
   }
 
-  hasArrayContainsFilter(): boolean {
-    return (
+  getRelationOpFilter(relationOp: RelationOp): RelationOp | null {
```

Comment:
This is a pretty unusual function contract, returning the relationOp back to you if it's found, else returning null.  Somebody looking at this in isolation would wonder why it doesn't just return a boolean.  I guess this unconventional contract is letting you be clever and save a bit of repetition in the calling code.  If we decided that was warranted, I'd recommend we put a comment here to provide that context...

But really, I think we can probably go back to returning a boolean here and still find a way to make the calling code clean / readable. :-)